src/detectors/BEMC/BEMC.cc

```diff
@@ -70,6 +83,28 @@ void InitPlugin(JApplication* app) {
       },
       app // TODO: Remove me once fixed
       ));
+  app->Add(new JOmniFactoryGeneratorT<CalorimeterPulseGeneration_factory>(
+      "EcalBarrelScFiPPulses", {"EcalBarrelScFiPAttenuatedHits"}, {"EcalBarrelScFiPPulses"},
+      {
+          .pulse_shape        = EcalBarrelScFi_pulse_shape,
+          .pulse_shape_params = EcalBarrelScFi_pulse_shape_params,
+          .edep_to_npe        = EcalBarrelScFi_edep_to_npe,
+          .timestep           = EcalBarrelScFi_timestep,
+          .ignore_thres       = EcalBarrelScFi_ignore_thres,
+      },
+      app // TODO: Remove me once fixed
+      ));
+  app->Add(new JOmniFactoryGeneratorT<CalorimeterPulseGeneration_factory>(
+      "EcalBarrelScFiNPulses", {"EcalBarrelScFiNAttenuatedHits"}, {"EcalBarrelScFiNPulses"},
+      {
+          .pulse_shape        = EcalBarrelScFi_pulse_shape,
+          .pulse_shape_params = EcalBarrelScFi_pulse_shape_params,
+          .edep_to_npe        = EcalBarrelScFi_edep_to_npe,
+          .timestep           = EcalBarrelScFi_timestep,
+          .ignore_thres       = EcalBarrelScFi_ignore_thres,
+      },
+      app // TODO: Remove me once fixed
+      ));
```

Comment:
This is great that you enable support for the barrel, but is this usable for other calorimeters. Ideally, we'd add this to every, but would be nice to demonstrate for at least one.